firestore/api/FirestoreTest/FirestoreTest.cs

```diff
@@ -0,0 +1,622 @@
+        {
+            return _getData.Run(args);
+        }
+
+        readonly CommandLineRunner _queryData = new CommandLineRunner()
+        {
+            VoidMain = QueryData.Main,
+            Command = "dotnet run"
+        };
+
+        protected ConsoleOutput RunQueryData(params string[] args)
+        {
+            return _queryData.Run(args);
+        }
+
+        readonly CommandLineRunner _orderLimitData = new CommandLineRunner()
+        {
+            VoidMain = OrderLimitData.Main,
+            Command = "dotnet run"
+        };
+
+        protected ConsoleOutput RunOrderLimitData(params string[] args)
+        {
+            return _orderLimitData.Run(args);
+        }
+
+        readonly CommandLineRunner _dataModel = new CommandLineRunner()
+        {
+            VoidMain = DataModel.Main,
+            Command = "dotnet run"
+        };
+
+        protected ConsoleOutput RunDataModel(params string[] args)
+        {
+            return _dataModel.Run(args);
+        }
+
+        readonly CommandLineRunner _transactionsAndBatchedWrites = new CommandLineRunner()
+        {
+            VoidMain = TransactionsAndBatchedWrites.Main,
+            Command = "dotnet run"
+        };
+
+        protected ConsoleOutput RunTransactionsAndBatchedWrites(params string[] args)
+        {
+            return _transactionsAndBatchedWrites.Run(args);
+        }
+
+        readonly CommandLineRunner _paginateData = new CommandLineRunner()
+        {
+            VoidMain = PaginateData.Main,
+            Command = "dotnet run"
+        };
+
+        protected ConsoleOutput RunPaginateData(params string[] args)
+        {
+            return _paginateData.Run(args);
+        }
+
+        private static async Task DeleteCollection(string collection)
+        {
+            FirestoreDb db = FirestoreDb.Create(Environment.GetEnvironmentVariable("FIRESTORE_PROJECT_ID"));
+            CollectionReference collectionReference = db.Collection(collection);
+            QuerySnapshot snapshot = await collectionReference.SnapshotAsync();
+            IReadOnlyList<DocumentSnapshot> documents = snapshot.Documents;
+            foreach (DocumentSnapshot document in documents)
+            {
+                await document.Reference.DeleteAsync();
+            }
+        }
+
+        // QUICKSTART TESTS
+        [Fact]
+        public void InitializeProjectIdTest()
+        {
+            var output = RunQuickstart("initialize-project-id", Environment.GetEnvironmentVariable("FIRESTORE_PROJECT_ID"));
+            Assert.Contains("Created Cloud Firestore client with project ID:", output.Stdout);
+        }
+
+        [Fact]
+        public void AddData1Test()
+        {
+            var output = RunQuickstart("add-data-1", Environment.GetEnvironmentVariable("FIRESTORE_PROJECT_ID"));
+            Assert.Contains("Added data to the alovelace document in the users collection.", output.Stdout);
+            DeleteCollection("users").Wait();
```

Comment:
The clean up operation needs to happen reliably even when an Assertion fails.  Make this class inherit from IDisposable and put the clean up code in the Dispose() method.